### PR TITLE
Sync lsp4jakarta issue #682 Diagnotics for Invalid Specialized annotation class

### PR DESCRIPTION
**Pull request by archana-1924**
lsp4jakarta issue: https://github.com/eclipse-lsp4jakarta/lsp4jakarta/issues/682
lsp4jakarta PR: https://github.com/eclipse-lsp4jakarta/lsp4jakarta/pull/939

### CODE REVIEW

File: src/main/resources/io/openliberty/tools/intellij/lsp4jakarta/messages/messages.properties [hunk 1]
```diff
@@ -139,6 +139,7 @@ InvalidWildcardTypeInInjectMethod = Wildcard types are not legal bean types. Inj
 InvalidWildcardTypeInProducerField = Wildcard types are not legal bean types. Producer fields must use concrete parameterized types without wildcards (?, ? extends, ? super).
 InvalidWildcardTypeInProducerMethod = Wildcard types are not legal bean types. Producer methods must return concrete parameterized types without wildcards (?, ? extends, ? super).
 InvalidNamedAnnotationOnNonFieldInjectionPoint = The @Named annotation on this injection point must specify a value. Only field injection points can omit the value (field name is assumed).
+InvalidSpecializesAnnotationOnNonBeanSuperclass = A bean annotated with @Specializes must directly extend the bean class of another CDI scope managed bean (e.g., @ApplicationScoped, @RequestScoped, @Dependent etc).
```
Comment (Joseph-Bineesh) [inline src/main/resources/io/openliberty/tools/intellij/lsp4jakarta/messages/messages.properties]: Try to refactor message to avoid eg, etc

**Review by Joseph-Bineesh — CHANGES_REQUESTED** [review]

**Review by Rejoice472 — COMMENTED** [review]
build is failing . could you please check this

**Comment by archana-1924** [comment]
> build is failing . could you please check this

Known issue: Moving a recent logic for specializes (https://github.com/OpenLiberty/liberty-tools-intellij/pull/1641) into this participant, the logics are getting conflicted and one is overwritten by other.

**Review by Joseph-Bineesh — DISMISSED** [review]
LGTM

File: src/test/resources/projects/maven/jakarta-sample/src/main/java/io/openliberty/sample/jakarta/cdi/SpecializesWithBeanSuperclass.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+package io.openliberty.sample.jakarta.cdi;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Specializes;
+
+/**
+ * Valid: BaseServiceWithScope is annotated with @ApplicationScoped (a valid CDI bean).
+ * @Specializes here should NOT trigger a diagnostic.
+ */
+@Specializes
```
Comment (Rejoice472) [inline src/test/resources/projects/maven/jakarta-sample/src/main/java/io/openliberty/sample/jakarta/cdi/SpecializesWithBeanSuperclass.java:10]: what will happen below scenarios?

1.) no super class
@Specializes
@ApplicationScoped
public class SpecializesWithBeanSuperclass  {
    @Override
    public String greet() { return "Custom Hello"; }
}

2.) implements interface

@Specializes
@ApplicationScoped
public class SpecializesWithBeanSuperclass implements BaseServiceInterface{
    @Override
    public String greet() { return "Custom Hello"; }
}

2.) implements interface and extends superclass

@Specializes
@ApplicationScoped
public class SpecializesWithBeanSuperclass extends BaseService implements BaseServiceInterface{
    @Override
    public String greet() { return "Custom Hello"; }
}

third scenario might be a edge case and won't be valid.
could you please add first two  in the test cases

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/CdiSpecializesDiagnosticsCollector.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+     * @param unit        the containing file
+     * @param diagnostics the list to add diagnostics to
+     */
+    private void validateSpecializes(PsiClass type, PsiJavaFile unit, List<Diagnostic> diagnostics) {
+        // Per CDI spec 3.1.4, only the direct (immediate) superclass must be a bean.
+        // A resolved superclass with any CDI scope annotation is valid; anything else
+        // (null superclass or unscoped superclass) is a definition error.
+        PsiClass superclass = type.getSuperClass();
+        boolean directSuperclassIsBean = superclass != null
+                && AnnotationUtils.hasAnyAnnotation(superclass, SCOPE_FQ_NAMES.toArray(String[]::new));
```
Comment (Rejoice472) [inline src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/CdiSpecializesDiagnosticsCollector.java]: Can classes annotated with a custom scope be treated as beans? Do we need to check that too?

**Review by Joseph-Bineesh — DISMISSED** [review]

**Review by Rejoice472 — DISMISSED** [review]

File: src/test/resources/projects/maven/jakarta-sample/src/main/java/io/openliberty/sample/jakarta/cdi/BaseServiceInterface.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+/*******************************************************************************
```
Comment (venmanyarun) [inline src/test/resources/projects/maven/jakarta-sample/src/main/java/io/openliberty/sample/jakarta/cdi/BaseServiceInterface.java]: remove copyrights from all test source files.. correct diagnostics range in tests based on new positions

**Review by venmanyarun — CHANGES_REQUESTED** [review]

**Review by Joseph-Bineesh — APPROVED** [review]
Re-approval

**Review by Rejoice472 — APPROVED** [review]

**Review by venmanyarun — APPROVED** [review]
